### ISSUE TITLE
URL parameter is added when the value is an empty string but not when it is an empty list

### ISSUE THREAD

**Issue by Lalee10** [psf/requests#6557]
Both an empty string `''` and an empty list `[]` are not of NoneType and are considered a falsy value and therefore should behave the same when being added as a URL parameter. But in requests when passing an empty string `''` as a parameter value the final URL contains the parameter but not when an empty list `[]` is passed as the value.

## Expected Result
When passing an empty list as parameter value, the final URL **should contain** the parameter.

## Actual Result
When passing an empty list as parameter value, the final URL **does not contain** the parameter.

## Reproduction Steps

```python
import requests

base_url = 'https://jsonplaceholder.typicode.com/todos'

params = {'_limit': 5, 'id__in': ''}
response = requests.get(base_url, params)
print(response.request.url)
# Output: https://jsonplaceholder.typicode.com/todos?_limit=5&id__in=

params = {'_limit': 5, 'id__in': []}
response = requests.get(base_url, params)
print(response.request.url)
# Output: https://jsonplaceholder.typicode.com/todos?_limit=5
```

## System Information

    $ python -m requests.help

```json
{
  "chardet": {
    "version": null
  },
  "charset_normalizer": {
    "version": "3.1.0"
  },
  "cryptography": {
    "version": ""
  },
  "idna": {
    "version": "3.4"
  },
  "implementation": {
    "name": "CPython",
    "version": "3.10.11"
  },
  "platform": {
    "release": "22.6.0",
    "system": "Darwin"
  },
  "pyOpenSSL": {
    "openssl_version": "",
    "version": null
  },
  "requests": {
    "version": "2.28.2"
  },
  "system_ssl": {
    "version": "1010115f"
  },
  "urllib3": {
    "version": "1.26.15"
  },
  "using_charset_normalizer": true,
  "using_pyopenssl": false
```

**Comment by Krupakar-Reddy-S** [comment]
Yes as observed this is the case when using any other empty datatype, except a **str()**, as it gives an empty string.

From Documentation:
*\" Note that any dictionary key whose value is None will not be added to the URL’s query string. \"* 

But as known, an empty list is not of None type, so i guess since it does not know what to do with empty datatypes it just treats then as None.

**Comment by amkarn258** [comment]
Proposed a fix here - https://github.com/psf/requests/pull/6561

**Comment by amritamishra01** [comment]
"Hi, I noticed PR #6561 has been inactive for over a year and lacks regression tests. I am going to pick this up, fix the logic to be Python 3 compliant, and importantly, add the missing test cases to ensure empty lists render as keys (e.g., ?key=). Submitting a PR shortly."

**Comment by nateprewitt** [comment]
I think this issue is conflating Python's semantics of truthiness with parameter behavior. `[]` is a list of *no* values and will not add anything to the the serialized parameters. `""` *is* a value saying we want an empty parameter. If you're trying to omit the parameter for some reason, you should be setting it None. This is working as intended.
